File: Javascript/Searching/BinarySearch/src/BinarySearch.ts
function search(arr: number[], target: number): number {
    
    let left = 0
    let right = arr.length - 1

    while(left <= right){
        let mid = Math.floor((left+right)/2);

        if(arr[mid] == target){
            return mid
        }

        else if(arr[mid] > target){
            right = mid - 1
        }

        else{
            left = mid + 1 
        }

    }
    
    return -1
}

let arr: number[] = [-2, 3, 4, 7, 8 ,9, 11, 13, 16, 17, 22, 26, 32]
let target: number = 17

let result = search(arr, target);

console.log(result)

export {}

File: Python/Searching/BinarySearch/BinarySearch.py
from math import floor

def search(arr: list, target: int) -> int:
    left = 0
    right = len(arr) - 1
    
    while(left <= right):
        mid = floor((left + right)/2)
        if(arr[mid] == target):
            return mid
        
        elif(arr[mid] < target):
            left = mid + 1
        else:
            right = mid -1
    
    return -1


if __name__ == '__main__':
    arr = [5,7,7,8,8,10]
    target = 8
    print(search(arr, target))
    
    